### PR TITLE
RFR: Prefix st2 processes with st2 for consistency

### PR DESCRIPTION
**Pull request by lakshmi-kannan**
Naming:

* All st2 components spun up as services are one worded. E.g.: st2sensorcontainer, st2api, st2rulesengine,...

* All scripts or tools use hyphenated words. E.g.: st2-rules-tester, st2-submit-debug-info, st2-register-content,...

### CODE REVIEW

File: docs/source/troubleshooting.rst [hunk 1]
```diff
@@ -88,7 +88,7 @@ the :ref:`testing-rules` section.
 You have now fixed your your rule criteria (or there was nothing wrong with it),
 but for some reason action is still not being fired / executed.
 
-If you see message similar to the one bellow in your rules_engine service log,
+If you see message similar to the one bellow in your rules engine service log,
```
Comment (DoriftoShoes) [inline docs/source/troubleshooting.rst]: 'below' one has one 'l'.

Comment (lakshmi-kannan) [inline docs/source/troubleshooting.rst]: lolol.

**Comment by DoriftoShoes** [comment]
Other than the typo (that apparently existed already, lol).  +1

File: docs/source/troubleshooting.rst (hunk 1 above)
Comment (lakshmi-kannan) [inline docs/source/troubleshooting.rst]: Who would have thought? 
https://github.com/StackStorm/st2/commit/647a865adfafe511ed59d73e6b15887f4aaee2f9

File: CHANGELOG.rst [hunk 2]
```diff
@@ -7,9 +7,10 @@ in development
 * Report a more user-friendly error if an action-chain task references an invalid or inexistent
   action. Also treat invalid / inexistent action as a top-level action-chain error. (improvement)
 * Report a more user-friendly error if an action-chain definition contains an invalid type.
-  (improvements)
+  (improvement)
 * Don't allow ``run-remote-script`` actions without an ``entry_point`` attribute - throw an
   exception when running an action. (improvement)
+* Rename all st2 processes to be prefixed by st2. (sensor_container is now st2sensorcontainer, rules_engine is now st2rulesengine, actionrunner is now st2actionrunner) (improvement)
```
Comment (Kami) [inline CHANGELOG.rst]: We should probably also have a separate "Upgrade notes" section in the documentation where we describe backward incompatible / breaking changes like this one.

Something like this - https://libcloud.readthedocs.org/en/latest/upgrade_notes.html

**Comment by lakshmi-kannan** [comment]
This isn't a backward incompatible change. When you upgrade the old processes are shut down and process with new names are started. st2ctl was adjusted to handle old process names. But yeah, having upgrade notes is useful.

**Comment by Kami** [comment]
Yeah, sorry, I didn't phrase myself correctly. I also meant changes like this which aren't necessary backward incompatible, but could affect some people.

**Comment by lakshmi-kannan** [comment]
Sure, I can open a PR.

**Pull request merged**